alghaibb/nyla
# Added contact page & more

## Overview
This PR introduces a new contact form feature for user inquiries and provides a mechanism for admins to view these submissions. Additionally, user experience enhancements such as toast notifications and form resetting have been implemented, along with a new contact icon in the footer for improved navigation.

## Changes
- Created `FormSubmission.ts` to define a new Payload CMS collection for storing contact form data.
- Implemented a new Contact page that includes a form with fields for name, email, message, and optional order ID.
- Added toast notifications to provide users with feedback upon form submission.
- Set up form resetting post-submission for a cleaner user experience.
- Placed a contact icon in the footer, linking directly to the Contact page for easy user access.

## Details
### `FormSubmission.ts`
- Defines a Payload CMS collection to capture user inquiries and store relevant information such as name, email, message, and order ID.
- Includes access control to allow admins to view these submissions.

### Contact Page
- Utilizes `react-hook-form` for form state management and `react-toastify` for user notifications.
- The form includes validation and error handling to guide users through the submission process.
- Upon successful submission, users receive a success notification, and the form resets.

### Footer Update
- A new contact icon has been added to the footer, acting as a shortcut to the Contact page, improving site navigation.

## Impact
- **User Interaction**: Users now have a dedicated and user-friendly way to contact the site administrators for inquiries or support.
- **Admin Efficiency**: Admins can easily access and manage user submissions through the Payload CMS admin interface.
- **Site Usability**: Enhancements to the footer and the immediate feedback provided by toast notifications significantly improve the overall user experience.

## Testing
- Manual testing was conducted on the Contact page to ensure that the form submits correctly and the user receives appropriate notifications.
- Verified that form submissions are accurately captured in the `FormSubmission` collection and viewable by admins.
- Checked the functionality of the new footer contact icon across various pages and screen sizes.

## Additional Notes
- Consider future enhancements such as automated email responses to users upon form submission.
- Integration of more complex validation for the order ID field might be beneficial.
- Feedback on the visual design and placement of the contact icon in the footer is welcome.

## How to Test
1. Visit the Contact page through the footer icon.
2. Fill out and submit the contact form.
3. Observe the toast notification and ensure the form resets after submission.
4. As an **admin**, navigate to the `FormSubmission` collection in Payload CMS to see the new entry.
